### PR TITLE
feat(native-filters): Highlight charts affected by focused native filter

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
When user focuses on a native filter (hovers over it or opens select menu), highlight all charts in scope of that native filter (by adding a box shadow around it) and hide the charts outside of scope (by setting opacity to 0.3).

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/118685945-d4d3e000-b803-11eb-9de2-6aaff0088e37.mov

### TEST PLAN
Set `DASHBOARD_NATIVE_FILTERS` feature flag to true.
Add some native filters to the dashboard - be creative with setting their scopes 🙂 
Verify that when you hover over Range or Time filter buttons, charts inside the scopes of those filters get highlighted and all other charts are less visible.
Verify that when you open a select menu of Select, Time Column, Time Grain and Group By filters, charts inside the scopes of those filters get highlighted and all other charts are less visible.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC: @junlincc @villebro @suddjian

### CODE REVIEW

**Comment by graceguo-supercat** [comment]
Could you create a sample dashboard with some Tabs, and set filter scope for those charts inside tab?

**Comment by junlincc** [comment]
@kgabryje i believe in the original implementation, when there are multiple tabs, tabs light up if any charts in the tab is mapped to a specific filter.

**Comment by kgabryje** [comment]
@graceguo-supercat @junlincc Thank you for comments! The charts inside of tabs are correctly highlighted, however if a chart in filter's scope is in an inactive tab, that tab doesn't get highlighted - I'll try to implement that.
@junlincc Are you sure that tabs were getting highlighted? I don't see the relevant code and there are no tabs highlights in the gif from Grace's comment here https://github.com/apache/superset/pull/10936#issuecomment-707224331

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardContainer.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ const DashboardContainer: FC<DashboardContainerProps> = ({ topLevelTabs }) => {
             activeKey={activeKey}
             renderTabBar={() => <></>}
             fullWidth={false}
-            animated
+            animated={false}
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/DashboardBuilder/DashboardContainer.tsx:75]: I disabled animation as it was breaking native filters - when user switched tab, `margin-left: 100%` style was set on tab content, which was covering native filters panel, causing the user to be unable to click native filters... And the animation looked ugly with native filters panel anyway 😛

**Comment by junlincc** [comment]
> Are you sure that tabs were getting highlighted?

got it, that was in the design but never implemented....😅 ;let's spend a day or 2 to add this feature in a seperate PR

- highlight other tabs that have charts scoped to the filter, when user is in a different tab. 
- when user switch to the highlighted tab, user see highlighted charts in tab

- please open a separate PR to disable animation in tab globally @ktmud 
- please open a separate PR to grey out filters that are not scoped to a specific tag.

**Comment by graceguo-supercat** [comment]
> @junlincc Are you sure that tabs were getting highlighted? I don't see the relevant code and there are no tabs highlights in the gif from Grace's comment here [#10936 (comment)](https://github.com/apache/superset/pull/10936#issuecomment-707224331)

For filter_box, we don't highlight tabs when only charts inside inactive tab will be impacted. We had this behavior is because filter_box could be placed inside tab. So that if a filter_box only has smaller tab scope, it should be placed inside the tab. It is possible that user sets a filter_box is applicable to multiple tabs but hide the filter_box inside a nested tab, but this case is small (why would someone do that?)

For build-in filter components, things are a little different: every filter component become globally visible in the dashboard, not matter it is applicable to single nested tab or multiple top-level tabs. This change might need us to think about the highlight behavior:

- When a filter component is applicable to a nested tab and currently user can not see the nested tab, filter component should be inactive (not in the current implementation but i assume this is in the near future), i think we still should highlight the scope even for the inactive filter component, so that user will know why the filter is not active now and what is its real scope. We should highlight the parent tab which is visible for the user when the charts are deep nested under other tabs.

- When a filter component is applicable to one more charts/tabs in the current active tab, filter component is active, we should highlight charts and tabs, so that behavior is consistent with above case.

File: superset-frontend/src/filters/components/GroupBy/types.ts [hunk 2]
```diff
@@ -40,6 +40,8 @@ export type PluginFilterGroupByProps = PluginFilterStylesProps & {
   behaviors: Behavior[];
   data: DataRecord[];
   setDataMask: SetDataMaskHook;
+  setFocusedFilter: () => void;
+  unsetFocusedFilter: () => void;
```
Comment (villebro) [inline superset-frontend/src/filters/components/GroupBy/types.ts:44]: Simplification idea: we could move the `setDataMask`, `setFocusedFilter` and `unsetFocusedFilter` into a shared interface/type `PluginFilterHooks` in `filters/components/types.ts` and then just `&` them into `PluginFilterGroupByProps` and the others.

File: superset-frontend/src/filters/components/common.ts [hunk 3]
```diff
@@ -21,8 +21,8 @@ import { Select } from 'src/common/components';
 import { PluginFilterStylesProps } from './types';
 
 export const Styles = styled.div<PluginFilterStylesProps>`
-  height: ${({ height }) => height};
-  width: ${({ width }) => width};
+  height: ${({ height }) => height}px;
+  width: ${({ width }) => width}px;
```
Comment (villebro) [inline superset-frontend/src/filters/components/common.ts:25]: oh one of these again 👍

**Review by villebro — APPROVED** [review]
A few non-blocking comments, LGTM!

File: superset-frontend/src/dashboard/actions/nativeFilters.ts [hunk 4]
```diff
@@ -116,7 +116,7 @@ type BootstrapData = {
   };
 };
 
-export interface SetBooststapData {
+export interface SetBootstrapData {
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/src/dashboard/actions/nativeFilters.ts:119]: 😆 Ah boost... I remember that I was so happy when C++11 (I think?) was released and they added a lot of stuff from boost to std 😄

File: superset-frontend/src/filters/components/GroupBy/types.ts (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/filters/components/GroupBy/types.ts]: Good point, done!

**Comment by junlincc** [comment]
@graceguo-supercat 

for current iteration - we will grey out filters that are not scoped to tabs from the filter bar, as just a visual suggestion that the specific filter won’t apply in the current scope(instead of disabling it). This should reach feature parity with filter box highlighting behavior 

For range filter, blue circle will be greyed out; for time range, color darker grey on hovering 
Screen Shot 2021-05-20 at 9 57 41 AM

Above change will be made in separate PRs 


We can discuss next iteration for grouping or foldering filter to tabs(but not now). 🙏

**Review by junlin-qa — APPROVED** [review]
approving a batch of native filter PRs, will test all as a whole ASAP.

**Comment by pseudoPixels** [comment]
This feature is very intuitive and useful for dashboard users. 
However, for some reason this feature is not working for our case (using Superset 1.2). Is there any special feature flag or so that we need to set to enable this? We do not get any error log/info, not sure whats we are missing. Can anyone please hint on it for possible issue? Many thanks all :)

**Comment by villebro** [comment]
> This feature is very intuitive and useful for dashboard users.
> However, for some reason this feature is not working for our case (using Superset 1.2). Is there any special feature flag or so that we need to set to enable this? We do not get any error log/info, not sure whats we are missing. Can anyone please hint on it for possible issue? Many thanks all :)

I don't believe this PR made it into 1.2 - however, this should be available in 1.3 (forthcoming)

**Pull request merged**